src/ConstructiveSolidGeometry/Transformations.jl

```diff
@@ -6,11 +6,11 @@ const Transformations{T} = NamedTuple{(:rotation, :translation), Tuple{SMatrix{3
 
 # Transformations{T}() where {T} = (rotation = one(SMatrix{3, 3, T, 9}), translation = zero(CartesianVector{T}))
 
-rotate(p::VP, r::AbstractMatrix) where {VP <: AbstractVolumePrimitive} = VP(p, origin = r * p.origin, rotation = r * p.rotation)
-(*)(r::AbstractMatrix, p::AbstractVolumePrimitive) = rotate(p, r)
+rotate(p::P, r::AbstractMatrix) where {P <: AbstractPrimitive} = P(p, origin = r * p.origin, rotation = r * p.rotation)
+(*)(r::AbstractMatrix, p::AbstractPrimitive) = rotate(p, r)
 
-translate(p::VP, v::CartesianVector) where {VP <: AbstractVolumePrimitive} = VP(p, origin = p.origin + v, rotation = p.rotation)
-(+)(p::AbstractVolumePrimitive, v::CartesianVector) = translate(p, v)
+translate(p::P, v::CartesianVector) where {P <: AbstractPrimitive} = P(p, origin = p.origin + v, rotation = p.rotation)
+(+)(p::AbstractPrimitive, v::CartesianVector) = translate(p, v)
```

Comment:
This will not work for `Polygon`, right?

Reply:
We can write a method for Polygon if we need it. 
Julia always uses the method with the most specific matching argument types :)